### DISCUSSION TITLE
Deserialization using the default deserializer

### DISCUSSION THREAD

**Discussion by pchanial** [wyfo/apischema#378] [Q&A]
After adding a deserializer to a class, the default deserializer is not accessible anymore. By default deserializer, I mean the one that can be used before adding a deserializer:
```python
import pytest
from apischema import ValidationError, deserialize, deserializer, schema
from apischema.conversions import Conversion, catch_value_error

@schema(min=1, max=2)
class Foo(int):
    pass

deserializer(Conversion(catch_value_error(Foo), source=str, target=Foo))
with pytest.raises(ValidationError) as exc:
    deserialize(Foo, 1)
assert exc.value.messages == ['expected type string, found integer']
```

When Foo derives from int, I can make it work by adding the following deserializer:
```python
deserializer(Conversion(catch_value_error(Foo), source=int, target=Foo))
```
and the str deserializer will check the constraints.

But what to do for dataclasses and still have the constraints checked by apischema ? The following fails since the default deserializer with dict is no more available:
```python
from dataclasses import dataclass
import pytest
from apischema import ValidationError, deserialize, deserializer, schema, validator
from apischema.conversions import Conversion, catch_value_error

@dataclass
class Foo:
    value: int

    @validator
    def check_value(self):
        if self.value != 42:
            raise ValidationError('Invalid value')

@deserializer
def from_int(value: int) -> Foo:
    return deserialize(Foo, {'value': value})

deserialize(Foo, 42)
```
So, shouldn't the default deserializer be kept ?

**Comment by wyfo** [comment]
I'm not sure to understand your question.

Default deserializer is not kept when a conversion is registered, because there must be a way to remove it, so I've chosen this design.
However, it's possible to ignore registered conversion and fall back on default deserializer, as explained in the [documentation](https://wyfo.github.io/apischema/0.17/conversions/#bypass-registered-conversion). Also, `identity` conversion can be registered itself in order to add the default deserializer back.

Does it solve your issue?
